File: features/document/src/main/java/com/nivasafinance/features/document/exception/S3DownloadException.java
package com.nivasafinance.features.document.exception;

import org.springframework.context.MessageSource;

import java.io.Serial;

public class S3DownloadException extends S3OperationException {
    
    @Serial
    private static final long serialVersionUID = 1L;
    
    public static final String S3_DOWNLOAD_FAILED_KEY = "error.document.s3.download.failed";
    
    public S3DownloadException(String message) {
        super(message);
    }
    
    public S3DownloadException(MessageSource messageSource) {
        super(S3_DOWNLOAD_FAILED_KEY, messageSource);
    }
    
    public S3DownloadException(String error, MessageSource messageSource) {
        super(S3_DOWNLOAD_FAILED_KEY, error, messageSource);
    }
}

File: features/document/src/main/java/com/nivasafinance/features/document/exception/S3DownloadException.kt
package com.nivasafinance.features.document.exception

import exception.BadRequestException
import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder
import java.util.Locale

class S3DownloadException(
    message: String
) : BadRequestException(message) {

    companion object {
        private val locale: Locale = LocaleContextHolder.getLocale()
        const val S3_DOWNLOAD_FAILED_KEY = "error.document.s3.download.failed"
    }

    constructor(messageSource: MessageSource) : this(
        messageSource.getMessage(S3_DOWNLOAD_FAILED_KEY, arrayOf(""), locale)
    )

    constructor(error: String, messageSource: MessageSource) : this(
        messageSource.getMessage(S3_DOWNLOAD_FAILED_KEY, arrayOf(error), locale)
    )
}
